exercises/dotnet/add-esdk-complete/src/DataModel.cs

```diff
@@ -0,0 +1,279 @@
+        }
+
+        public static DocumentBundle FromData(byte[] data)
+        {
+            return FromDataAndContext(data, new Dictionary<string, string>());
+        }
+
+        public static DocumentBundle FromDataAndPointer(byte[] data, PointerItem pointer)
+        {
+            return new DocumentBundle(data, pointer);
+        }
+
+        public static DocumentBundle FromDataAndContext(byte[] data, Dictionary<string, string> context)
+        {
+            return new DocumentBundle(data, PointerItem.Generate(context));
+        }
+
+        public override string ToString()
+        {
+            return $"{pointer}: {Encoding.ASCII.GetString(data)}";
+        }
+    }
+
+    public class GuidKey
+    {
+        private Guid _key;
+
+        public GuidKey()
+        {
+            _key = Guid.NewGuid();
+        }
+
+        public GuidKey(string guid)
+        {
+            _key = Guid.Parse(guid);
+        }
+
+        public override string ToString()
+        {
+            return _key.ToString();
+        }
+    }
+
+    public class BaseItem
+    {
+        protected readonly static string PartitionKeyName = Config.PartitionKeyName;
+        protected readonly static string SortKeyName = Config.SortKeyName;
+
+        private readonly AttributeValue _partitionKey;
+        private readonly AttributeValue _sortKey;
+
+        public AttributeValue PartitionKey => _partitionKey;
+        public AttributeValue SortKey => _sortKey;
+
+        public BaseItem(string partitionKey, string sortKey)
+        {
+            _partitionKey = new AttributeValue(partitionKey);
+            _sortKey = new AttributeValue(sortKey);
+        }
+
+        public virtual Dictionary<string, AttributeValue> ToItem()
+        {
+            return new Dictionary<string, AttributeValue>
+            {
+                { PartitionKeyName, _partitionKey },
+                { SortKeyName, SortKey }
+            };
+        }
+
+        public override bool Equals(object? obj)
+        {
+            var otherItem = obj as BaseItem;
+            if (otherItem is null)
+            {
+                return false;
+            }
+            return _partitionKey == otherItem._partitionKey && SortKey == otherItem.SortKey;
+        }
+
+        public override int GetHashCode()
+        {
+            return HashCode.Combine(_partitionKey, SortKey);
+        }
+
+        public override string ToString()
+        {
+            return $"PartitionKey: {_partitionKey.S}, SortKey: {_sortKey.S}";
+        }
+    }
+
+    public class ContextItem : BaseItem
+    {
+        static private readonly string _prefix = Config.ObjectTarget;
+
+        public ContextItem(string contextKey, GuidKey objectTarget) : base(contextKey, objectTarget.ToString()) { }
+
+        public static string Canonicalize(string key)
+        {
+            if (!key.StartsWith(_prefix))
+            {
+                return key;
+            }
+            return _prefix + key;
```

Comment:
I believe the condition is flipped here and throughout:

```suggestion
            if (key.StartsWith(_prefix))
            {
                return key;
            }
            return _prefix + key;
```